src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -19,9 +19,15 @@
 // For each hosting API, we define a function prototype and a function pointer
 // The prototype is useful for implicit linking against the dynamic coreclr
 // library and the pointer for explicit dynamic loading (dlopen, LoadLibrary)
+#ifdef __cplusplus
 #define CORECLR_HOSTING_API(function, ...) \
     extern "C" int CORECLR_CALLING_CONVENTION function(__VA_ARGS__); \
```

Comment:
```suggestion
    EXTERN_C int CORECLR_CALLING_CONVENTION function(__VA_ARGS__); \
```

I think you can just use the `EXTERN_C` macro here.

Reply:
I'd need to define it in the specific file since it's standalone. That has a risk of collision with other code.